### PR TITLE
Contribute p/z specific test code to midstream repos

### PR DESCRIPTION
**Pull request by AlstonDias**
CASC changes for single Jenkins integration.

### CODE REVIEW

File: openshift/e2e-common.sh [hunk 1]
```diff
@@ -100,18 +100,30 @@ function run_e2e_rekt_tests(){
   echo "Replacing images used in Rekt test resources with the images built in CI"
 
   echo "Replacing knative-eventing-test-event-library image"
-  sed -i -e "s|registry.ci.openshift.org/openshift/knative-.*:knative-eventing-test-event-library|${KNATIVE_EVENTING_TEST_EVENT_LIBRARY}|g" "$(dirname "$0")/../test/rekt/resources/eventlibrary/eventlibrary.yaml"
+  sed -i -e "s|registry.ci.openshift.org/openshift/knative-.*:knative-eventing-test-event-library|quay.io/openshift-knative/eventing/event-library:v1.6|g" "$(dirname "$0")/../test/rekt/resources/eventlibrary/eventlibrary.yaml"
```
Comment (pierDipi) [inline openshift/e2e-common.sh:103]: The env variables are used in CI to inject the dynamically built images.

Can't you define

```bash
export KNATIVE_EVENTING_TEST_EVENT_LIBRARY="quay.io/openshift-knative/eventing/event-library:v1.6"
# ... others test images env vars

source openshift/e2e-common.sh
```

/hold

File: test/rekt/resources/broker/broker.go [hunk 2]
```diff
@@ -58,7 +58,7 @@ func init() {
 }
 
 func WithEnvConfig() []manifest.CfgFn {
-	cfg := []manifest.CfgFn{WithBrokerClass(EnvCfg.BrokerClass)}
+	cfg := []manifest.CfgFn{WithBrokerClass("MTChannelBasedBroker")}
```
Comment (maschmid) [inline test/rekt/resources/broker/broker.go:61]: this should not be necessary, you should be able to `export BROKER_CLASS=MTChannelBasedBroker` before running the tests to force MTChannelBasedBroker if necessary.

File: openshift/e2e-common.sh [hunk 3]
```diff
@@ -100,18 +100,30 @@ function run_e2e_rekt_tests(){
...
+  local test_name="${1:-}"
+  local run_command=""
+  local failed=0
+ 
+  if [ -n "$test_name" ]; then
+      local run_command="-run ^(${test_name})$"
+  fi
+  # check for test flags
+  local run_flags="-timeout=90m -parallel=20"
+  if [ -v EVENTING_TEST_FLAGS ] && [ ! -z "$EVENTING_TEST_FLAGS" ]; then
```
Comment (mgencur) [inline openshift/e2e-common.sh:123]: I think this complicated condition could be converted to
` if [ -n EVENTING_TEST_FLAGS ]; then`

File: openshift/e2e-common.sh [hunk 4]
```diff
@@ -100,18 +100,30 @@ function run_e2e_rekt_tests(){
...
-  go_test_e2e -timeout=1h -parallel=20 ./test/rekt || failed=$?
+  local test_name="${1:-}"
+  local run_command=""
+  local failed=0
+ 
+  if [ -n "$test_name" ]; then
+      local run_command="-run ^(${test_name})$"
+  fi
+  # check for test flags
+  local run_flags="-timeout=90m -parallel=20"
```
Comment (mgencur) [inline openshift/e2e-common.sh:122]: The flags are a "string" here but on line 148 it's an array so you're using two different approaches.
The ideal solution is to use arrays but if you don't care about shellcheck (I don't mind) you could use just strings everywhere.

File: openshift/e2e-common.sh [hunk 5]
```diff
@@ -132,11 +144,23 @@ function run_e2e_tests(){
...
-  go_test_e2e -timeout=50m -parallel=20 ./test/e2e \
+  # check for test flags
+  RUN_FLAGS=("-timeout=50m -parallel=20")
+  if [ -n "${EVENTING_TEST_FLAGS:-}" ]; then
+    RUN_FLAGS="${EVENTING_TEST_FLAGS}"
+  fi
+
+  # check for test args
+  if [ -n "${EVENTING_TEST_ARGS:-}" ]; then
+    common_opts=("${EVENTING_TEST_ARGS} --kubeconfig $KUBECONFIG")
```
Comment (mgencur) [inline openshift/e2e-common.sh:155]: Why is KUBECONFIG passed only when EVENTING_TEST_ARGS are defined? Sounds like it should be passed as part of those EVENTING_TEST_ARGS when calling the script. Then you can avoid passing --kubeconfig on this line.

File: openshift/e2e-common.sh [hunk 6]
```diff
@@ -154,11 +178,23 @@ function run_conformance_tests(){
...
-  go_test_e2e -timeout=30m -parallel=12 ./test/conformance \
+  # check for test flags
+  RUN_FLAGS=("-timeout=50m -parallel=20")
+  if [ -n "${EVENTING_TEST_FLAGS:-}" ]; then
+    RUN_FLAGS="${EVENTING_TEST_FLAGS}"
+  fi
+
+  # check for test args
+  if [ -n "${EVENTING_TEST_ARGS:-}" ]; then
+    common_opts=("${EVENTING_TEST_ARGS} --kubeconfig $KUBECONFIG")
```
Comment (mgencur) [inline openshift/e2e-common.sh:189]: Same as above. Passing KUBECONFIG seems redundant here.

**Review by mgencur — CHANGES_REQUESTED** [review]
Hi @AlstonDias , there are some parts in the script that should be fixed. 

Could you please provide some background on what you're trying to achieve? If I understand it correctly you need this:
* be able to run a specific test
* use multiarch images
* use different parallel/timeout?

**Comment by AlstonDias** [comment]
Hi @mgencur, I have done all code changes as mentioned in the above comments and tried the test. It is all working.

File: openshift/e2e-common.sh [hunk 7]
```diff
@@ -111,7 +111,19 @@ function run_e2e_rekt_tests(){
...
-  go_test_e2e -timeout=1h -parallel=20 ./test/rekt || failed=$?
+  local test_name="${1:-}"
+  local run_command=""
+  local failed=0
+ 
+  if [ -n "$test_name" ]; then
+      local run_command="-run ^(${test_name})$"
+  fi
+  # check for test flags
+  RUN_FLAGS=("-timeout=90m -parallel=20")
```
Comment (mgencur) [inline openshift/e2e-common.sh:122]: You're using an array of one element here. And if EVENTING_TEST_FLAGS is defined the value will overwrite the single element in the array, two lines below.
Not sure it's intentional, the array is useless here but it works. Using just plain string would be ok as well:
```
RUN_FLAGS="-timeout=50m -parallel=20"
if [ -n "${EVENTING_TEST_FLAGS:-}" ]; then
  RUN_FLAGS="${EVENTING_TEST_FLAGS}"
fi
...
...
go_test_e2e ${RUN_FLAGS} ./test/rekt
```

File: openshift/e2e-common.sh [hunk 8]
```diff
@@ -132,11 +144,23 @@ function run_e2e_tests(){
...
-  go_test_e2e -timeout=50m -parallel=20 ./test/e2e \
+  # check for test flags
+  RUN_FLAGS=("-timeout=50m -parallel=20")
+  if [ -n "${EVENTING_TEST_FLAGS:-}" ]; then
+    RUN_FLAGS="${EVENTING_TEST_FLAGS}"
+  fi
+
+  # check for test args
+  if [ -n "${EVENTING_TEST_ARGS:-}" ]; then
+    common_opts=("${EVENTING_TEST_ARGS}")
```
Comment (mgencur) [inline openshift/e2e-common.sh:155]: You're assigning an array to a variable that was defined as "string" earlier. Perhaps better to use just `common_opts="${EVENTING_TEST_ARGS}"`

**Review by mgencur — COMMENTED** [review]
Hey @AlstonDias it looks much better. My latest comments are just nits. I think it works but it might work by accident.

File: openshift/e2e-common.sh (hunk 7 above)
Comment (AlstonDias) [reply openshift/e2e-common.sh:122]: So you want me replace the array to string value everywhere in the code?

Comment (mgencur) [reply openshift/e2e-common.sh]: Yes, please. On lines that you modified.

File: openshift/e2e-common.sh [hunk 9]
```diff
@@ -154,11 +178,23 @@ function run_conformance_tests(){
       local run_command="-run ^(${test_name})$"
   fi
 
-  go_test_e2e -timeout=30m -parallel=12 ./test/conformance \
+  # check for test flags
+  RUN_FLAGS="-timeout=50m -parallel=20"
```
Comment (mgencur) [inline openshift/e2e-common.sh:182]: I didn't notice it previously but you're changing parallel to 20 which might cause issues in CI. Please have the default value "-timeout=30m -parallel=12" as you can override it via EVENTING_TEST_FLAGS anyway. Thanks

**Pull request merged**